packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -190,103 +421,38 @@ export default class VrackMoveDialogCtrl {
-          }),
-          managedBareMetal: allServicesParam.dedicatedCloud.filter(
-            (service) => {
-              return (
-                VrackMoveDialogCtrl.isServiceAllowed(service) &&
-                service.productReference === 'MBM'
-              );
-            },
-          ),
-          dedicatedServer: allServicesParam.dedicatedServer.filter(
-            VrackMoveDialogCtrl.isServiceAllowed,
-          ),
-          dedicatedServerInterface: allServicesParam.dedicatedServerInterface.filter(
-            ({ dedicatedServer }) =>
-              VrackMoveDialogCtrl.isServiceAllowed(dedicatedServer),
-          ),
-        };
-        allServices = mapValues(allServices, (services, serviceType) => {
-          if (isArray(services)) {
-            return map(services, (service) =>
-              this.constructor.fillServiceData(serviceType, service),
-            );
-          }
-          return services;
-        });
-
-        // We need to append dedicatedServerInterfaces list to dedicatedServers list.
-        if (
-          has(allServices, 'dedicatedServerInterface') &&
-          allServices.dedicatedServerInterface.length > 0
-        ) {
-          // If dedicatedServers list doesn't exist, we create it first.
-          if (!has(allServices, 'dedicatedServer')) {
-            allServices.dedicatedServer = [];
-          }
-
-          angular.forEach(
-            allServices.dedicatedServerInterface,
-            (serverInterface) => {
-              allServices.dedicatedServer.push(serverInterface);
-            },
-          );
-
-          allServices.dedicatedServerInterface = [];
-        }
-
-        return this.getAvailableServices(allServices);
-      });
-  }
-
-  static isServiceAllowed(service) {
-    return (
-      !service.expired &&
-      [STATUS.ok, STATUS.delivered].includes(service.state) &&
-      service.status === STATUS.ok
-    );
-  }
-
   getAvailableServices(allServices) {
     return pickBy(allServices, (_, serviceType) => {
       return this.features.isFeatureAvailable(FEATURE_NAMES[serviceType]);
     });
   }
 
+  /**
+   * get the services already present in the selected vrack
+   * @returns {*}
+   */
   getVrackServices() {
     return this.OvhApiVrack.Aapi()
       .services({ serviceName: this.serviceName })
       .$promise.then((allServicesParam) => {
         let allServices = allServicesParam;
         allServices = mapValues(allServices, (servicesParams, serviceType) => {
           let services = servicesParams;
-          if (isArray(services)) {
-            services = map(services, (service) =>
+          if (Array.isArray(services) && services.length) {
+            services = services.map((service) =>
               this.constructor.fillServiceData(serviceType, service),
             );
           }
           return services;
         });
 
-        if (has(allServices, 'dedicatedCloud')) {
+        if (has(allServices, SERVICES_NAMES['DCLOUD_SERVICE'])) {
```

Comment:
```suggestion
        if (has(allServices, SERVICES_NAMES.DCLOUD_SERVICE)) {
```
and same for other in same conttroller